core/commands/UserUpdateCommand.php

```diff
@@ -0,0 +1,345 @@
+require_api( 'constant_inc.php' );
+require_api( 'config_api.php' );
+require_api( 'helper_api.php' );
+require_api( 'user_api.php' );
+
+use Mantis\Exceptions\ClientException;
+
+require_once( dirname( __FILE__ ) . '/../../api/soap/mc_account_api.php' );
+require_once( dirname( __FILE__ ) . '/../../api/soap/mc_api.php' );
+
+/**
+ * A command that updates a user account.
+ *
+ * Note that only fields to be updated need to be provided. If extra fields
+ * are provided and it matches current state, it will be ignored.
+ *
+ * Sample:
+ * {
+ *   "query": {
+ *     "user_id": 1
+ *   },
+ *   "payload": {
+ *     "user": {
+ *       "name": "vboctor",
+ *       "real_name": "Victor Boctor",
+ *       "email": "vboctor@example.com",
+ *       "access_level": { "name" => "developer" },
+ *       "enabled": true,
+ *       "protected": false
+ *     },
+ *     "notify_user": false
+ *   }
+ * }
+ */
+class UserUpdateCommand extends Command {
+	/**
+	 * @var integer user id
+	 */
+	private $user_id;
+
+	/**
+	 * @var string user name
+	 */
+	private $username;
+
+	/**
+	 * @var string user real name
+	 */
+	private $realname;
+
+	/**
+	 * @var string user email address
+	 */
+	private $email;
+
+	/**
+	 * @var integer access level
+	 */
+	private $access_level;
+
+	/**
+	 * @var boolean user can edit their information.  Usually, true for shared accounts.
+	 */
+	private $protected;
+
+	/**
+	 * @var boolean user account enabled for login.
+	 */
+	private $enabled;
+
+	/**
+	 * @var boolean notify user of account changes via email.
+	 */
+	private $notify_user;
+
+	/**
+	 * @var array old user information
+	 */
+	private $old_user;
+
+	/**
+	 * @var array new user information
+	 */
+	private $new_user;
+
+	/**
+	 * Constructor
+	 *
+	 * @param array $p_data The command data.
+	 */
+	function __construct( array $p_data ) {
+		parent::__construct( $p_data );
+	}
+
+	/**
+	 * Validate the data.
+	 */
+	function validate() {
+		# Get id of target user
+		$this->user_id = helper_parse_id( $this->query( 'user_id' ), 'user_id' );
+
+		# Get info about logged in user which can be different than target user
+		$t_actor_is_admin = current_user_is_administrator();
+		$t_actor_can_manage_users = access_has_global_level( config_get( 'manage_user_threshold' ) );
+		$t_actor_user_id = auth_get_current_user_id();
+
+		# Ensure user exists
+		if( !user_exists( $this->user_id ) ) {
+			throw new ClientException( 'User does not exist', ERROR_USER_NOT_FOUND );
+		}
+
+		# User Data
+		$t_user = $this->payload( 'user' );
+		if( is_null( $t_user ) ) {
+			throw new ClientException( 'Missing user data', ERROR_EMPTY_FIELD, array( 'user' ) );
+		}
+
+		# Protected
+		$t_old_protected = user_is_protected( $this->user_id );
+		$t_new_protected = isset( $t_user['protected'] ) ? (int)$t_user['protected'] : null;
+		if( !is_null( $t_new_protected ) && $t_old_protected !== $t_new_protected ) {
+			$this->protected = $t_new_protected ? true : false;
+		}
```

Comment:
Not sure why you're assigning int|null to $t_new_protected, when the finality is to compare it with a bool returned by user_is_protected() (which would btw always fail due to the strict-type compare bool vs int) and finally assign true/false to the class property.

This should be simplified, something like (not tested)
```php
if( isset( $t_user['protected'] ) ) {
    $t_new_protected = (bool)$t_user['protected'];
    if( $t_old_protected != $t_new_protected ) {
        $this->protected = $t_new_protected;
    }
}
```

Reply:
I have change int to bool and removed strict equality. This style is consistent with the rest of the code.